src/main/java/org/apache/pirk/querier/wideskies/encrypt/EncryptQuery.java

```diff
@@ -178,11 +217,14 @@ public Querier encrypt(int numThreads) throws InterruptedException, PIRException
       {
         // Hash collision
         selectorQueryVecMapping.clear();
-        hashKey = queryInfo.getHashKey() + ++keyCounter;
+        hashKey = hashKeyBase + getRandByteString(10);
         logger.debug("index = " + index + "selector = " + selector + " hash collision = " + hash + " new key = " + hashKey);
         index = -1;
       }
     }
+    
+    // Save off the final hashKey that we ended up using
+    queryInfo.setHashKey(hashKey);
```

Comment:
Seems reasonable to me - I don't think that we need to allow a user to specify a key. If it becomes a requirement down the line, we can always add it back as an option.